lib/Ix/DBIC/ResultSet.pm

```diff
@@ -411,10 +411,16 @@ sub ix_create ($self, $ctx, $to_create) {
     } else {
       my $exception = $@;
 
-      my $error;
+      my ($error, $res);
```

Comment:
As discussed on Slack, I think that it should be the above eval (now to become try/catch) that should return these two, rather than doing it in the no-$row branch.  The way done here, you've missed logging the creation id, and other row-reporting logic.